core/store/src/adapter/trie_store.rs

```diff
@@ -100,31 +91,21 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
         Self { store_update: StoreUpdateHolder::Reference(store_update) }
     }
 
-    /// Reads shard_uid mapping for given shard.
-    /// If the mapping does not exist, it means that `shard_uid` maps to itself.
-    fn read_shard_uid_mapping_from_db(
-        &self,
-        shard_uid: ShardUId,
-    ) -> Result<ShardUId, StorageError> {
-        let store = TrieStoreAdapter::new(Store::new(self.store_update.storage.clone()));
-        store.read_shard_uid_mapping_from_db(shard_uid)
-    }
-
     pub fn decrement_refcount_by(
         &mut self,
         shard_uid: ShardUId,
         hash: &CryptoHash,
         decrement: NonZero<u32>,
     ) {
         let mapped_shard_uid =
-            self.read_shard_uid_mapping_from_db(shard_uid).expect("decrement_refcount_by failed");
+            read_shard_uid_mapping_from_db(&self.store_update().store, shard_uid);
```

Comment:
Can you keep it as a method on self?

Reply:
@wacban  Does https://github.com/near/nearcore/commit/ac1751f119263adfaf81523f2739afc9b2f08e2b do what you mean?


I think there is one more thing to do:
```
impl<'a> TrieStoreUpdateAdapter<'a> {
    fn get_key_from_shard_uid_and_hash(&self, shard_uid: ShardUId, hash: &CryptoHash) -> [u8; 40] {
        self.store_update.store.trie_store().get_key_from_shard_uid_and_hash(shard_uid, hash)
    }
```
Here, calling `trie_store()` indirectly clones store. I would like to create `StoreHolder` that would allow to pass store by reference, analogously how `StoreUpdateHolder` works. @shreyan-gupta wdyt?

Reply:
Ah that's getting more complicated instead of less. Feel free to keep as is to keep it simple.

Reply:
Avoided cloning in https://github.com/near/nearcore/pull/12232/commits/432a96d1b0643e10ee398d18c6368d1efa76796d